src/VirtualMonoRepo/InitializeVMR.proj

```diff
@@ -0,0 +1,122 @@
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Project>
+  <PropertyGroup>
+    <ArtifactsDir>$(RepoRoot)artifacts/</ArtifactsDir>
+    <VirtualMonoRepoTasksAssembly>$(ArtifactsDir)bin/VirtualMonoRepo.Tasks/$(Configuration)/net6.0/VirtualMonoRepo.Tasks.dll</VirtualMonoRepoTasksAssembly>
+
+    <VmrDir Condition=" '$(VmrDir)' == '' ">$(ArtifactsDir)vmr/</VmrDir>
+    <VmrDir>$([MSBuild]::EnsureTrailingSlash('$(VmrDir)'))</VmrDir>
+    <TmpDir Condition=" '$(TmpDir)' == '' ">$(RepoRoot)tmp/</TmpDir>
+    <VmrDefaultBranch Condition=" '$(VmrDefaultBranch)' == '' ">main</VmrDefaultBranch>
+
+    <!-- We set this for the source build targets to work well -->
+    <TarballDir>$(VmrDir)</TarballDir>
+    <InitializeVMR>true</InitializeVMR>
+  </PropertyGroup>
+
+  <Import Project="../SourceBuild/Arcade/tools/SourceBuildArcadeTarball.targets" />
+
+  <PropertyGroup>
+    <VmrGitInfoDir>$(TarballGitInfoDir)</VmrGitInfoDir>
+  </PropertyGroup>
+
+  <UsingTask TaskName="Microsoft.DotNet.VirtualMonoRepo.Tasks.VirtualMonoRepo_Initialize" AssemblyFile="$(VirtualMonoRepoTasksAssembly)" />
+
+  <Target Name="Build" DependsOnTargets="InitializeVMR" />
+
+  <!--
+    This target initializes a fresh VMR from scratch by assembling it similarly to the tarball.
+    Instead of cloning the repositories, we pull the sources via the `darc vmr initialize` command.
+    Most of the targets are reused from the tarball generation process (SourceBuildArcadeTarball.targets).
+  -->
+  <Target Name="InitializeVMR" DependsOnTargets="GenerateFullNuGetVersion;
+                                                 GenerateVersionFile;
+                                                 InitializeCleanVmr;
+                                                 CopyTarballContent;
+                                                 CommitInitialContent;
+                                                 SetupSelfGithubInfo;
+                                                 InitializeRepoAndDependentsRecursive;
+                                                 CommitGitInfoFiles;
+                                                 CopyTextOnlyPackages;
+                                                 CommitTextOnlyPackages">
+    <Message Text="VMR was successfully initialized in '$(VmrDir)'" Importance="High" />
+  </Target>
+
+  <Target Name="InitializeCleanVmr">
+    <Error Text="VmrDir not specified" Condition=" '$(VmrDir)' == '' " />
```

Comment:
Is this even possible with the defaulting logic?

Reply:
Fair point. I've made `VmrDir` compulsory with no default so that we are sure it doesn't point somewhere else in the recursive invocations of the .proj file and similar